Persephone/MPDClient/MPDClient.swift

```diff
@@ -12,285 +12,43 @@ import mpdclient
 class MPDClient {
   var delegate: MPDClientDelegate?
 
-  private var connection: OpaquePointer?
-  private var isConnected: Bool = false
-  private var status: Status?
-  private var queue: [Song] = []
+  var connection: OpaquePointer?
+  var isConnected: Bool = false
+  var isIdle: Bool = false
+  var status: Status?
+  var queue: [Song] = []
 
-  private let commandQueue = DispatchQueue(label: "commandQueue")
+  let commandQueue = OperationQueue()
+  var commandsQueued: UInt = 0
 
   enum Command {
-    case prevTrack, nextTrack, playPause, stop,
-      fetchStatus, fetchQueue, fetchAllAlbums
-  }
-
-  struct Idle: OptionSet {
-    let rawValue: UInt32
-
-    static let database = Idle(rawValue: 0x1)
-    static let storedPlaylist = Idle(rawValue: 0x2)
-    static let queue = Idle(rawValue: 0x4)
-    static let player = Idle(rawValue: 0x8)
-    static let mixer = Idle(rawValue: 0x10)
-    static let output = Idle(rawValue: 0x20)
-    static let options = Idle(rawValue: 0x40)
-    static let update = Idle(rawValue: 0x80)
-    static let sticker = Idle(rawValue: 0x100)
-    static let subscription = Idle(rawValue: 0x200)
-    static let message = Idle(rawValue: 0x400)
+    case prevTrack, nextTrack, playPause, stop, seekCurrentSong,
+      fetchStatus, fetchQueue, playTrack, fetchAllAlbums,
+      playAlbum, getAlbumURI
   }
 
   init(withDelegate delegate: MPDClientDelegate?) {
+    commandQueue.maxConcurrentOperationCount = 1
     self.delegate = delegate
   }
 
-  func connect(host: String, port: Int) {
-    commandQueue.async { [unowned self] in
-      guard let connection = mpd_connection_new(host, UInt32(port), 10000),
-        mpd_connection_get_error(connection) == MPD_ERROR_SUCCESS
-        else { return }
-
-      self.isConnected = true
-
-      guard let status = mpd_run_status(connection)
-        else { return }
-
-      self.connection = connection
-      self.status = Status(status)
-
-      self.fetchQueue()
-      self.fetchAllAlbums()
-      self.idle()
-
-      self.delegate?.didConnect(mpdClient: self)
-      self.delegate?.didUpdateState(mpdClient: self, state: self.status!.state)
-      self.delegate?.didUpdateTime(mpdClient: self, total: self.status!.totalTime, elapsedMs: self.status!.elapsedTimeMs)
-      self.delegate?.didUpdateQueue(mpdClient: self, queue: self.queue)
-      self.delegate?.didUpdateQueuePos(mpdClient: self, song: self.status!.song)
-    }
-  }
-
-  func disconnect() {
+  func queueCommand(
```

Comment:
"Queue command" reads like a noun phrase to me, rather than a verb/command: what do you think about naming this "<em>en</em>queue command"?